### PR TITLE
Disable deep hashing if memoization is off

### PR DESCRIPTION
**Pull request by philippjfr**
Currently ``Callable`` runs deep hashing on the stream values whether or not memoization is enabled. This PR simply disables it when memoization is off, which has no effect but can be more efficient.

### CODE REVIEW

**Comment by jlstevens** [comment]
Simple change that does make sense.

**Comment by philippjfr** [comment]
>Simple change that does make sense.

Unfortunately it breaks some things. I'll have to think more about it.

**Comment by philippjfr** [comment]
Okay figured it out, the ``memoize`` parameter was being abused by modifying it. I've now added an internal attribute that is used for this purpose instead and the ``Callable.memoize`` is now used solely to control whether memoization is enabled or not.

**Comment by jlstevens** [comment]
Sounds like the right fix. 

That said it might be nice to give the concept a different name than ``self._memoize`` that helps express the other purposes you mention - otherwise it isn't clear why there is an underscore attribute of the same name that is distinct from the parameter.

**Comment by jlstevens** [comment]
How about ``_stream_memoization`` as distinct from the state on the ``Callable``?

Other than that, I'm happy to merge once the tests pass (ideally with some new unit tests!)

File: holoviews/streams.py [hunk 1]
```diff
@@ -404,11 +443,11 @@ class PointerXY(LinkedStream):
     the plot bounds, the position values are set to None.
     """
 
-    x = param.ClassSelector(class_=(Number, util.basestring), default=None,
+    x = param.ClassSelector(class_=(Number, util.basestring, tuple), default=None,
```
Comment (philippjfr) [inline holoviews/streams.py]: Unrelated change that snuck in. Allows the Pointer stream to work with nested categorical axes so worth keeping (and I can't be bothered to make a new PR).

File: holoviews/core/util.py [hunk 2]
```diff
@@ -101,7 +101,7 @@ def default(self, obj):
         elif isinstance(obj, np.ndarray):
             return obj.tolist()
         if pd and isinstance(obj, (pd.Series, pd.DataFrame)):
-            return repr(sorted(list(obj.to_dict().items())))
+            return obj.to_csv().encode('utf-8')
```
Comment (philippjfr) [inline holoviews/core/util.py]: A fair bit faster, although still not great, hence also adding a hashkey.

**Comment by philippjfr** [comment]
Ready to merge.

**Comment by jlstevens** [comment]
Looks good. Merging.

**Pull request merged**